include/openthread/platform/psa.h

```diff
@@ -0,0 +1,237 @@
+ * @retval OT_ERROR_INVALID_ARGS  @p aInput or @p aOutput was set to NULL.
+ *
+ */
+otError otPlatPsaImportKey(psa_key_id_t *        aKeyId,
+                           psa_key_type_t        aKeyType,
+                           psa_algorithm_t       aKeyAlgorithm,
+                           psa_key_usage_t       aKeyUsage,
+                           psa_key_persistence_t aKeyPersistence,
+                           const uint8_t *       aKey,
+                           size_t                aKeyLen);
+
+/**
+ * Export a key stored in PSA ITS.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for crypto operations.
+ * @param[out]  aBuffer           Pointer to the buffer where key needs to be exported.
+ * @param[in]   aBufferLen        Length of the buffer passed to store the exported key.
+ * @param[out]  aKeyLen           Length of the exported key.
+ *
+ * @retval OT_ERROR_NONE          Successfully exported  @p aKeyId.
+ * @retval OT_ERROR_FAILED        Failed to export @p aInput.
+ * @retval OT_ERROR_INVALID_ARGS  @p aBuffer was NULL
+ *
+ */
+otError otPlatPsaExportKey(psa_key_id_t aKeyId, uint8_t *aBuffer, uint8_t aBufferLen, size_t *aKeyLen);
+
+/**
+ * Destroy a key stored in PSA ITS.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for crypto operations.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ *
+ */
+otError otPlatPsaDestroyKey(psa_key_id_t aKeyId);
+
+/**
+ * Export public key of the keypair stored in ITS.
+ *
+ * @param[in]   aKeyId            Reference to the key to be exported.
+ * @param[out]  aOutput           Pointer to the buffer to store exported public key.
+ * @param[in]   aOutputSize       Size of the output buffer passed.
+ * @param[out]  aOutputLen        Length of the exported key.
+ *
+ * @retval OT_ERROR_NONE          Successfully exported  @p aKeyId.
+ * @retval OT_ERROR_FAILED        Failed to export @p aKeyId.
+ * @retval OT_ERROR_INVALID_ARGS  @p aOutput or @p aOutputLen was NULL
+ *
+ */
+otError otPlatPsaExportPublicKey(psa_key_id_t aKeyId, uint8_t *aOutput, size_t aOutputSize, size_t *aOutputLen);
+
+/**
+ * Sign a message hash using a key stored.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for Signing.
+ * @param[in]   aKeyAlgorithm     Key algorithm compatible with the key stored.
+ * @param[in]   aHash             Hash of the message to be signed.
+ * @param[in]   aHashSize         Size of the hash.
+ * @param[out]  aSignature        Buffer to store the calculated signature.
+ * @param[in]   aSignatureSize    Size of the buffer passed.
+ * @param[in]   aSignatureLen     Length of the signature calculated.
+ *
+ * @retval OT_ERROR_NONE          Successfully signed  @p aHash.
+ * @retval OT_ERROR_FAILED        Failed to sign @p aHash.
+ *
+ */
+otError otPlatPsaSignHash(psa_key_id_t    aKeyId,
```

Comment:
This seems to be unused. Do we need it, since we have equivalent PSA API?

Reply:
This is part of the code implemented for ECDH, I think it missed the cleanup. I will remove it from the commit.